compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/ReshapeFusion.cpp

```diff
@@ -652,6 +652,93 @@ struct DropScatterUnitIndexDepth final : public OpRewritePattern<ScatterOp> {
   }
 };
 
+struct DropGatherBatchUnitDims final : public OpRewritePattern<GatherOp> {
+  using OpRewritePattern<GatherOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(GatherOp gatherOp,
+                                PatternRewriter &rewriter) const override {
+
+    if (gatherOp.getBatchRank() <= 1) {
+      return rewriter.notifyMatchFailure(
+          gatherOp, "dropping unit dims from batch rank 1 is not supported.");
+    }
```

Comment:
nit: remove the first blank line.

> In particular, don't put more than one or two blank lines between functions, resist starting functions with a blank line, don't end functions with a blank line, and be sparing with your use of blank lines. 

https://google.github.io/styleguide/cppguide.html#Vertical_Whitespace